### ISSUE TITLE
use constraints to place a cut

### ISSUE THREAD

**Issue by ndaman** [CadQuery/cadquery#1199]
Is it possible to use constraints to place objects for a cut operation? Or else some kind of undocumented way to "subtract" instead of "add" in an Assembly?

**Comment by Jojain** [comment]
No, the concept of an assembly is to position finished parts, so if you need to remove material from a shape you should do it before the assembly step.

**Comment by ndaman** [comment]
Is there a way to use constraints outside of the assembly? They are often
more convenient and reliable than translate and rotate operations.

**Comment by adam-urbanczyk** [comment]
So as @Jojain mentioned this is not currently possible, but it crossed my mind to eventually implement some editing operations.

You could extract the locations from the assy object (`cq.Assembly.loc`), apply the on the individual components and then cut.

**Comment by ndaman** [comment]
Cool, thanks!
